mypy.ini

```diff
@@ -0,0 +1,22 @@
+[mypy]
```

Comment:
Should the `mypi.ini` configs go in the [`tox.ini`](https://github.com/sqlfluff/sqlfluff/blob/master/tox.ini) instead?

Reply:
I would love for it to be in `tox.ini` too but it mypy does not read config from `tox.ini` :( 

There was a [discussion in a PR](https://github.com/python/mypy/pull/2761) in mypy that was merged but the support for `tox.ini` was not included in the end :( 

![Screenshot 2020-10-05 at 13 43 04](https://user-images.githubusercontent.com/4448469/95080929-bad00c00-0710-11eb-80e6-f0d8fc256a11.png)

Reply:
Bummer! @alanmcruickshank are you ok with having a separate `mypy.ini`?

Reply:
if it's not supported in `tox.ini` then yes 👍